### DISCUSSION TITLE
Shell script based actions often fails with <action_name>: No such file or directory

### DISCUSSION THREAD

**Discussion by vblagoje** [community/community#108611] [Actions]
### Select Topic Area

Question

### Body

Hey guys,

In my custom action, I'm using shell to run it, and it is pretty simple, yet it fails 60-70% of the time; the rest of the time, it works. I've only come here after deep desperation of being unable to figure it out.

The action shell script is at https://github.com/vblagoje/create-or-update-release-note/blob/main/action.yml

Fail trace is https://github.com/vblagoje/workflow-playground/actions/runs/7806654062/job/21293472554?pr=55
Success trace is https://github.com/vblagoje/workflow-playground/actions/runs/7918049403/job/21615564457

I am open to any suggestions; it doesn't have to be that Shell is used as the action runner. Anything that works is ok.

TIA

**Comment by vblagoje** [comment]
The issue stems from backticks in the content I'm trying to put into the file. Using:
```printf '%s\n' "${{ inputs.note-content }}" > "$FILENAME"```
works, but it removes these strings surrounded with backticks :-( 

Do you have any ideas on correctly writing this text content to a file from bash without running into these issues?

**Comment by CoreTiQ** [comment]
printf '%s\n' "${inputs.note-content}" > "$FILENAME"

**Comment by vblagoje** [comment]
I tried it and I get:
``` /home/runner/work/_temp/0e4423cd-781e-4d0b-baae-69d74485d508.sh: line 30: ${inputs.note-content}: bad substitution```
Details at https://github.com/vblagoje/workflow-playground/actions/runs/7942825152/job/21686514835?pr=70

**Comment by vblagoje** [comment]
I rewrote the GA with javascript. No issues any more :-)

**Reply by CoreTiQ** [reply]
Sorry for the late reply, I was busy yesterday.
I'm glad you solved the problem
